Add tap_like() test function for regex matching

larchunix opened the pull request:
Feel free to merge it if you find it useful.

andrewgregory commented:
This looks pretty good.  If you're willing to make a couple small changes, I'll be happy to merge it.  First, rename `tap_rematch` to `tap_like` to more closely follow perl's Test::More module which this is modelled after.  Second, I would like the output to be different from tap_is_\* to make it clear that this is a regex comparison.  The related Test::More function replaces the got/expected format with:

```
#                   'foo'
#     doesn't match '(?^:bar)'
```

I'm not especially fond of that format, so I would also accept just replacing the quotes in the expected line with `/`:

```
#          got: 'baz'
#     expected: /^\<(foo|bar)\>/
```

Your choice.

larchunix committed:
Add tap_like() test function for regex matching

v2: rename tap_rematch() to tap_like()
    enclose regex with / in diagnostic output

Signed-off-by: Denis Ollier <larchunix@users.noreply.github.com>

larchunix commented:
Thanks for your feedback.

I have revised the PR according to your remarks.

andrewgregory commented:
Merged manually.
